conans/test/integration/environment/test_env.py

```diff
@@ -389,3 +387,66 @@ def generate(self):
     assert "MYVAR2: PkgAValue2 PkgCValue2 PkgBValue2 PkgDValue2!!!" in client.out
     assert "MYVAR3: PkgDValue3 PkgBValue3 PkgCValue3 PkgAValue3!!!" in client.out
     assert "MYVAR4: PkgDValue4!!!" in client.out
+
+
+def test_environment_scripts_generated_envvars():
+    consumer_pkg = textwrap.dedent(r"""
+        from conans import ConanFile
+        from conan.tools.env import VirtualBuildEnv, VirtualRunEnv
+        class Pkg(ConanFile):
+            settings = "os"
+            requires = "require_pkg/1.0"
+            build_requires = "build_require_pkg/1.0"
+            generators = "VirtualRunEnv", "VirtualBuildEnv"
+        """)
+
+    client = TestClient()
+    conanfile = (GenConanfile().with_package_file("bin/myapp", "myexe")
+                               .with_package_file("lib/mylib", "mylibcontent")
+                               .with_settings("os"))
+    client.save({"build_require_pkg/conanfile.py": conanfile,
+                 "require_pkg/conanfile.py": conanfile,
+                 "consumer_pkg/conanfile.py": consumer_pkg})
+
+    client.run("export build_require_pkg build_require_pkg/1.0@")
+    client.run("export require_pkg require_pkg/1.0@")
+
+    client.run("install consumer_pkg --build")
+    if platform.system() == "Windows":
+        conanbuildenv = client.load("conanbuildenv.bat")
+        conanrunenv = client.load("conanrunenv.bat")
+        assert "LD_LIBRARY_PATH" not in conanbuildenv
+        assert "LD_LIBRARY_PATH" not in conanrunenv
+    else:
+        conanbuildenv = client.load("conanbuildenv.sh")
+        conanrunenv = client.load("conanrunenv.sh")
+        assert "LD_LIBRARY_PATH" in conanbuildenv
+        assert "LD_LIBRARY_PATH" in conanrunenv
+
+    # Build context LINUX - Host context LINUX
+    client.run("install consumer_pkg -s:b os=Linux -s:h os=Linux --build")
```

Comment:
Better be sure to remove the previous generated script files, or to split the test or something. If something was not working, it would read the script file from the previous calls.